acme/http.go

```diff
@@ -33,9 +34,11 @@ func httpHead(url string) (resp *http.Response, err error) {
 	req.Header.Set("User-Agent", userAgent())
 
 	resp, err = http.DefaultClient.Do(req)
-	if resp.Body != nil {
-		resp.Body.Close()
+	if err != nil {
+		return resp, err
 	}
+	io.Copy(ioutil.Discard, resp.Body)
```

Comment:
yeah i have no strong preference, i prefer the explicitness (like i said; adding as a habit); the **bug** was referencsing `resp.Body` and a nil deref when `.Do()` returned an `err` and `resp` was nil.